### DISCUSSION TITLE
About the License of Ipopt

### DISCUSSION THREAD

**Discussion by yusukekobori** [coin-or/Ipopt#635] [Q&A] [answered]
I am trying to commercially utilize a product that has solved a convex quadratic programming problem using Ipopt.

As I am not familiar with open-source software, I would like some help.

Ipopt is under the EPL2.0 license, but compiling it as Ipopt.exe requires third-party libraries/solvers with different licenses, according to Github:
・HSL
・ASL
・MUMPS
・Pardiso

So, my understanding is that using Ipopt may result in not being solely covered by the EPL2.0 license, depending on the use of these third-party components, correct?

**Comment by svigerske** [comment, marked as answer]
Yes, you need to check also the licenses of the dependencies that make it into your product.
Mumps is open-source. ASL should not be a problem either. For HSL and Pardiso you would need to acquire separate licenses.
Note that not all dependencies are required (https://coin-or.github.io/Ipopt/index.html#PREREQUISITES). For example, Ipopt needs only one linear solver.
